UM/Math/Matrix.py

```diff
@@ -209,6 +411,89 @@ def setPerspective(self, fovy, aspect, near, far):
         self._data[2, 3] = -1.
         self._data[3, 2] = (2. * far * near) / (near - far)
 
+
+    def decompose(self):
+        """Return sequence of transformations from transformation matrix.
+
+        matrix : array_like
+            Non-degenerative homogeneous transformation matrix
+
+        Return tuple of:
+            scale : vector of 3 scaling factors
+            shear : list of shear factors for x-y, x-z, y-z axes
+            angles : list of Euler angles about static x, y, z axes
+            translate : translation vector along x, y, z axes
+
+        Raise ValueError if matrix is of wrong type or degenerative.
+
+        >>> T0 = translation_matrix([1, 2, 3])
+        >>> scale, shear, angles, trans, persp = decompose_matrix(T0)
```

Comment:
You removed perspective from the return values, so this documentation is outdated. Twice below, too.